components/dashboard/src/components/UsageBasedBillingConfig.tsx

```diff
@@ -90,14 +96,31 @@ export default function UsageBasedBillingConfig({ attributionId, hideSubheading
     }, [attributionId, refreshSubscriptionDetails]);
 
     useEffect(() => {
+        const params = new URLSearchParams(location.search);
+        setStripeParams({
+            setupIntentId: params.get("setup_intent") || undefined,
+            redirectStatus: params.get("redirect_status") || undefined,
+        });
+    }, [location.search]);
```

Comment:
If I understand correctly, these params will be there when the page reloads when stripe redirects back. I think we could treat this slightly different, and instead of setting state when location.search changes, we could run an effect when component mounts, and have it check for the params, and call `subscribeToStripe()` if they're present. I think this would remove some of the unpredictability around when we call `subscribeToStripe()`. So something like:

```ts
useEffect(() => {
    const params = new URLSearchParams(location.search);
    const setupIntentId = params.get("setup_intent");
    const redirectStatus = params.get("redirect_status");
    
    if (setupIntentId && redirectStatus) {
        subscribeToStripe();
    }
// eslint-disable-next-line react-hooks/exhaustive-deps don't want to run after component mounts
}, []);
```
We could then remove the `useEffect(subscribeToStripe, [subscribeToStripe]);` below.

Reply:
Doing that on mount (`[]` as deps and checking for params) should be a clean solution, which would also work well with a following `history.replaceState` to remove those. If one would then hit reload, assuming the location got already cleaned, we would not hit a second API call 👍🏻 


Adding an additional guard (local to the page, like `didAlreadyCallSubscribe`) might still make sense though, because of [strict](https://react.dev/reference/react/StrictMode) mode at least 🤷🏻‍♂️ .

Reply:
Thx, I changed as advised, and it works nicely. :lotus_position: